rbac/migration_tool/migrate.py

```diff
@@ -0,0 +1,130 @@
+    """Generate a set of relationships for the given set of v2 role bindings."""
+    relationships = set[Relationship]()
+    for v2_role_binding in v2_role_bindings:
+        relationships.add(
+            Relationship(
+                "role_binding",
+                v2_role_binding.id,
+                "granted",
+                "role",
+                v2_role_binding.role.id,
+            )
+        )
+        # Bind directly to resource for now
+        # relationships.add(
+        #    Relationship("workspace", "org_migration_root", "user_grant", "role_binding", v2_role_binding.id))
+        for perm in v2_role_binding.role.permissions:
+            relationships.add(Relationship("role", v2_role_binding.role.id, perm, "user", "*"))
+        for bound_resource in v2_role_binding.resources:
+            parent_relation = "parent" if bound_resource.resource_type == "workspace" else "workspace"
+            relationships.add(
+                Relationship(
+                    bound_resource.resource_type,
+                    bound_resource.resourceId,
+                    parent_relation,
+                    "workspace",
+                    "org_migration_root",
+                )
+            )
+            relationships.add(
+                Relationship(
+                    bound_resource.resource_type,
+                    bound_resource.resourceId,
+                    "user_grant",
+                    "role_binding",
+                    v2_role_binding.id,
+                )
+            )
+    return relationships
+
+
+def stringify_spicedb_relationship(rel: Relationship):
+    """Stringify a relationship for logging."""
+    return (
+        rel.resource_type + ":" + rel.resource_id + "#" + rel.relation + "@" + rel.subject_type + ":" + rel.subject_id
+    )
+
+
+def migrate_role(role: Role):
+    """Migrate a role from v1 to v2."""
+    v1_roles = extract_info_into_v1_role(role)
+    # With the replicated role bindings algorithm, role bindings are scoped by group, so we need to add groups
+    # TODO: replace the hard coded groups
+    groups = frozenset(
+        {
+            V1group("a_team", frozenset({"user_1"})),
+            V1group("b_team", frozenset({"user_2"})),
+        }
+    )
+    v1_roles = [dataclasses.replace(r, groups=groups) for r in v1_roles]
+
+    # This is where we wire in the implementation we're using into the Migrator
+    v1_to_v2_mapping = shared_system_role_replicated_role_bindings_v1_to_v2_mapping
+    permissioned_role_migrator = Migrator(v1_to_v2_mapping)
+    v2_roles = [v2_role for v1_role in v1_roles for v2_role in permissioned_role_migrator.migrate_v1_roles(v1_role)]
+    spicedb_rel_summary = spicedb_relationships(frozenset(v2_roles))
+    for rel in spicedb_rel_summary:
+        logger.info(stringify_spicedb_relationship(rel))
+
+
+def migrate_roles_for_tenant(tenant: Tenant):
+    """Migrate all roles for a given tenant."""
+    for role in tenant.role_set.all():
```

Comment:
As we chatted about in the collab, it might be helpful to have a `ROLE_MIGRATION_APP_EXCLUSION_LIST` or similar to filter on here, where permissions on roles with that list of apps could be excluded from the migration.

Reply:
I updated it with command parameter input